thirdeye-ui/src/app/components/rca/anomaly-time-series-card/anomaly-time-series-card.utils.tsx

```diff
@@ -82,85 +85,100 @@ export const generateSeriesDataForEvaluation = (
+                      gradient: {
+                          from: Palette.COLOR_VISUALIZATION_STROKE_CURRENT,
+                          to: Palette.COLOR_VISUALIZATION_STROKE_CURRENT,
+                          fromOpacity: 0.1,
+                          toOffset: 25,
+                          toOpacity: 0,
+                      },
+                      color: Palette.COLOR_VISUALIZATION_STROKE_CURRENT,
+                      stroke: Palette.COLOR_VISUALIZATION_STROKE_CURRENT,
+                      strokeWidth: Dimension.WIDTH_VISUALIZATION_STROKE_CURRENT,
+                      data: timeSeriesData.current
+                          .map((value, idx) => {
+                              return {
+                                  y: 0,
+                                  y1: value,
+                                  x: timeSeriesData.timestamp[idx],
+                              };
+                          })
+                          .filter((d) => Number.isFinite(d.y)),
+                      tooltip: {
+                          pointFormatter: (dataPoint: DataPoint): string =>
+                              formatLargeNumberV1(
+                                  (dataPoint as ThresholdDataPoint).y1
+                              ),
+                      },
+                      legendIcon: (svgBound: number, color: string) => {
+                          return (
+                              <line
+                                  stroke={color}
+                                  strokeWidth={10}
+                                  x1="0"
+                                  x2={`${svgBound}`}
+                                  y1={`${svgBound / 2}`}
+                                  y2={`${svgBound / 2}`}
+                              />
+                          );
+                      },
+                  },
+              ]
+            : []),
+        ...(!hidePredicted
+            ? [
+                  {
+                      name: translation("label.predicted"),
+                      type: SeriesType.LINE,
+                      color: Palette.COLOR_VISUALIZATION_STROKE_BASELINE,
+                      strokeWidth:
+                          Dimension.WIDTH_VISUALIZATION_STROKE_BASELINE,
+                      data: timeSeriesData.expected
+                          .map((value, idx) => {
+                              return {
+                                  y: value,
+                                  x: timeSeriesData.timestamp[idx],
+                              };
+                          })
+                          .filter((d) => Number.isFinite(d.y)),
+                      tooltip: {
+                          pointFormatter: (d: DataPoint): string =>
+                              formatLargeNumberV1(d.y),
+                      },
+                      strokeDasharray: `${Dimension.DASHARRAY_VISUALIZATION_BASELINE}`,
+                  },
+              ]
+            : []),
```

Comment:
Refactor this to something like:

```tsx
export const generateSeriesDataForEvaluation = (
    alertEvaluation: AlertEvaluation,
    filteredAlertEvaluations: [AlertEvaluation, AnomalyFilterOption[]][],
    translation: (id: string) => string,
    hideUpperLowerBound?: boolean,
    hideActivity?: boolean,
    hidePredicted?: boolean
): Series[] => {
    const filteredTimeSeriesData: Series[] = filteredAlertEvaluations.map(
        (alertEvalAndFilters) => {
            const [filteredAlertEvaluation, filterOptions] =
                alertEvalAndFilters;
            const filteredAlertEvaluationTimeSeriesData =
                filteredAlertEvaluation.detectionEvaluations
                    .output_AnomalyDetectorResult_0.data;

            return {
                name: filterOptions.map(concatKeyValueWithEqual).join(" & "),
                type: SeriesType.LINE,
                data: filteredAlertEvaluationTimeSeriesData.current
                    .map((value, idx) => {
                        return {
                            y: value,
                            x: filteredAlertEvaluationTimeSeriesData.timestamp[
                                idx
                                ],
                        };
                    })
                    .filter((d) => Number.isFinite(d.y)),
                tooltip: {
                    valueFormatter: formatLargeNumberV1,
                },
            };
        }
    );

    const timeSeriesData =
        alertEvaluation.detectionEvaluations.output_AnomalyDetectorResult_0
            .data;

    const chartSeries: Series[] = [];

    if (!hideUpperLowerBound) {
        chartSeries.push({
            enabled: false,
            name: translation("label.upper-and-lower-bound"),
            type: SeriesType.AREA_CLOSED,
            fillOpacity: 0.1,
            color: Palette.COLOR_VISUALIZATION_STROKE_UPPER_AND_LOWER_BOUND,
            data: timeSeriesData.lowerBound.map((value, idx) => {
                return {
                    y: value,
                    y1: timeSeriesData.upperBound[idx],
                    x: timeSeriesData.timestamp[idx],
                };
            }),
            stroke: Palette.COLOR_VISUALIZATION_STROKE_UPPER_AND_LOWER_BOUND,
            strokeWidth: Dimension.WIDTH_VISUALIZATION_STROKE_DEFAULT,
            tooltip: {
                pointFormatter: (d: DataPoint): string =>
                    formatLargeNumberV1(d.y),
            },
            legendIndex: 10,
        })
    }

    if (!hideActivity) {
        chartSeries.push({
            name: translation("label.activity"),
            type: SeriesType.AREA_CLOSED,
            gradient: {
                from: Palette.COLOR_VISUALIZATION_STROKE_CURRENT,
                to: Palette.COLOR_VISUALIZATION_STROKE_CURRENT,
                fromOpacity: 0.1,
                toOffset: 25,
                toOpacity: 0,
            },
            color: Palette.COLOR_VISUALIZATION_STROKE_CURRENT,
            stroke: Palette.COLOR_VISUALIZATION_STROKE_CURRENT,
            strokeWidth: Dimension.WIDTH_VISUALIZATION_STROKE_CURRENT,
            data: timeSeriesData.current
                .map((value, idx) => {
                    return {
                        y: 0,
                        y1: value,
                        x: timeSeriesData.timestamp[idx],
                    };
                })
                .filter((d) => Number.isFinite(d.y)),
            tooltip: {
                pointFormatter: (dataPoint: DataPoint): string =>
                    formatLargeNumberV1(
                        (dataPoint as ThresholdDataPoint).y1
                    ),
            },
            legendIcon: (svgBound: number, color: string) => {
                return (
                    <line
                        stroke={color}
                        strokeWidth={10}
                        x1="0"
                        x2={`${svgBound}`}
                        y1={`${svgBound / 2}`}
                        y2={`${svgBound / 2}`}
                    />
                );
            },
        })
    }

    if (!hidePredicted) {
        chartSeries.push({
            name: translation("label.predicted"),
            type: SeriesType.LINE,
            color: Palette.COLOR_VISUALIZATION_STROKE_BASELINE,
            strokeWidth:
            Dimension.WIDTH_VISUALIZATION_STROKE_BASELINE,
            data: timeSeriesData.expected
                .map((value, idx) => {
                    return {
                        y: value,
                        x: timeSeriesData.timestamp[idx],
                    };
                })
                .filter((d) => Number.isFinite(d.y)),
            tooltip: {
                pointFormatter: (d: DataPoint): string =>
                    formatLargeNumberV1(d.y),
            },
            strokeDasharray: `${Dimension.DASHARRAY_VISUALIZATION_BASELINE}`,
        })
    }

    return [
        ...chartSeries,
        ...filteredTimeSeriesData,
    ];
};
```

^^ that's based on the current file in this PR but there are some upstream changes that need to be merged